src/node/waku_node.py

```diff
@@ -84,6 +109,46 @@ def start(self, **kwargs):
             logger.error(f"REST service did not become ready in time: {ex}")
             raise
 
+    @retry(stop=stop_after_delay(5), wait=wait_fixed(0.1), reraise=True)
+    def register_rln(self, **kwargs):
+        logger.debug("Registering RLN credentials...")
+        self._docker_manager.create_network()
+        self._ext_ip = self._docker_manager.generate_random_ext_ip()
+        self._ports = self._docker_manager.generate_ports()
+        self._rest_port = self._ports[0]
+        self._tcp_port = self._ports[1]
+        self._websocket_port = self._ports[2]
+        self._discv5_port = self._ports[3]
+        self._metrics_port = self._ports[4]
+        self._api = REST(self._rest_port)
+        self._volumes = []
+
+        default_args = {
+            "rln-creds-id": None,
+            "rln-creds-source": None,
+        }
+
+        for key, value in kwargs.items():
+            key = key.replace("_", "-")
+            default_args[key] = value
+
+        rln_args, rln_creds_set, keystore_path = self.parse_rln_credentials(default_args, True)
+
+        if rln_creds_set:
+            self._container = self._docker_manager.start_container(
+                self._docker_manager.image, self._ports, rln_args, self._log_path, self._ext_ip, self._volumes
+            )
+
+            logger.debug(f"Executed container from image {self._image_name}. REST: {self._rest_port} to register RLN")
+
+            delay(15)
+            self.stop()
```

Comment:
I think we need to add a try-except-finally block and use stop in the finally so it gets executed always

Reply:
Resolved by adding exception rise to  rln_credential_store_ready(). stop() turned out to be trimming container from keystore in cache when delay too short. delay() removed as well.